diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -0,0 +1,114 @@
+import com.pinterest.ktlint.core.ast.ElementType.WHEN_CONDITION_WITH_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.WHEN_ENTRY
+import com.pinterest.ktlint.core.ast.children
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.psi.psiUtil.siblings
+import org.jetbrains.kotlin.utils.addToStdlib.ifNotEmpty
+
+/**
+ * [1] Enumerations (In another rule)
+ * [2] Value arguments
+ * [3] Class properties and parameters
+ * [4] Function value parameters
+ * [5] Parameters with optional type (including setters)
+ * [6] Indexing suffix
+ * [7] Lambda parameters
+ * [8] when entry
+ * [9] Collection literals (in annotations)Type arguments
+ * [10] Type arguments
+ * [11] Type parameters
+ * [12] Destructuring declarations
+ */
+class TrailingCommaRule(private val configRules: List<RulesConfig>) : Rule("trailing-comma") {
+    private var isFixMode: Boolean = false
+    private val configuration by lazy {
+        TrailingCommaConfiguration(
+            this.configRules.getRuleConfig(TRAILING_COMMA)?.configuration ?: emptyMap()
+        )
+    }
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val commonConfig by configRules.getCommonConfiguration()
+        if (commonConfig.kotlinVersion >= ktVersion) {
+            val (type, config) = when (node.elementType) {
+                VALUE_ARGUMENT_LIST -> Pair(VALUE_ARGUMENT, configuration.getParam("valueArgument"))
+                VALUE_PARAMETER_LIST -> Pair(VALUE_PARAMETER, configuration.getParam("valueParameter"))
+                INDICES -> Pair(REFERENCE_EXPRESSION, configuration.getParam("referenceExpression"))
+                WHEN_ENTRY -> Pair(WHEN_CONDITION_WITH_EXPRESSION, configuration.getParam("whenConditions"))
+                COLLECTION_LITERAL_EXPRESSION -> Pair(STRING_TEMPLATE, configuration.getParam("collectionLiteral"))
+                TYPE_ARGUMENT_LIST -> Pair(TYPE_PROJECTION, configuration.getParam("typeArgument"))
+                TYPE_PARAMETER_LIST -> Pair(TYPE_PARAMETER, configuration.getParam("typeParameter"))
+                DESTRUCTURING_DECLARATION -> Pair(
+                    DESTRUCTURING_DECLARATION_ENTRY,
+                    configuration.getParam("destructuringDeclaration")
+                )
+                else -> return
+            }
+            val astNode = node
+                .children()
+                .toList()
+                .ifNotEmpty { this.lastOrNull { it.elementType == type } }
+            astNode?.checkTrailingComma(config)
+        }
+    }
+
+    private fun ASTNode.checkTrailingComma(config: Boolean) {
+        val isNextComma = this.siblings(true).map { it.elementType }.contains(COMMA)
```

Comment:
We should also check if there is a line break (or comment) after this node. Only then should we suggest to use trailing comma. It won't make sense to have, e.g., `fun foo(a: Int,) {}`